File: 287-find-the-duplicate-number/287-find-the-duplicate-number.java
class Solution {
    public int findDuplicate(int[] nums) {
        boolean[] count = new boolean[100005];
        
        for(int num : nums)
        {
            if(count[num])
                return num;
            else
                count[num] = true;
        }
        
        return 0;
    }
}

File: Solutions/287-find-the-duplicate-number/287-find-the-duplicate-number.cpp
class Solution {
public:
    int findDuplicate(vector<int>& nums) {
        int fast = 0, slow = 0;
        
        fast = nums[nums[fast]];
        slow= nums[slow];
        
        while(nums[fast] != nums[slow])
        {
            fast = nums[nums[fast]];
            slow= nums[slow];
        }
        
        fast = 0;
        while(nums[fast] != nums[slow])
        {
            fast = nums[fast];
            slow= nums[slow];
        }
        
        return nums[fast];
    }
};